src/server/listener.cc

```diff
@@ -210,7 +210,7 @@ Listener::run() {
 
         int ready_fds = poller.poll(events, 128, std::chrono::milliseconds(-1));
         if (ready_fds == -1) {
-            if (errno == EINTR && g_listen_fd == -1) return;
+            if (errno == EINTR) continue;
```

Comment:
Yes, the logic is that EINTR can occur in any case during an epoll_wait, and I don't see why the handling thereof shiould differ from one case to another. In practice, this caused a useless exception for me when executing the augmentor test under gdb, which was solved by the above.